Makefile

```diff
@@ -1,25 +1,68 @@
+ROOT_DIR := $(shell dirname $(realpath $(firstword $(MAKEFILE_LIST))))
+OUTPUT_DIR := $(ROOT_DIR)/output
+BUILD_DIR := $(OUTPUT_DIR)/build
+IMAGE_DIR := $(OUTPUT_DIR)/images
+INITRAMFS_DIR := $(OUTPUT_DIR)/initramfs
 
-all: buildroot/Makefile buildroot/.config
-	make -C buildroot
+BR_INIT_ENV := BR2_EXTERNAL=$(ROOT_DIR)
 
+
+.PHONY: all
+all: $(IMAGE_DIR)/rootfs.img $(IMAGE_DIR)/uImage $(IMAGE_DIR)/am335x-kno_txt.dtb
+
+.PHONY: clean
 clean:
-	BR2_EXTERNAL=.. make -C buildroot clean
+	rm -rf $(OUTPUT_DIR)
+
+.PHONY: prepare-structure
+prepare-structure:
+	mkdir -p $(OUTPUT_DIR)
+	mkdir -p $(IMAGE_DIR)
+	mkdir -p $(INITRAMFS_DIR)
+
+.PHONY: prepare
+prepare: prepare-structure $(BUILD_DIR)/rootfs/.config $(BUILD_DIR)/initramfs/.config
+
+.PHONY: source
+source: rootfs-source initramfs-source
 
-buildroot/Makefile:
+.PHONY: rootfs-source
+rootfs-source: $(BUILD_DIR)/rootfs/.config
+	$(MAKE) -C $(BUILD_DIR)/rootfs source
+
+.PHONY: initramfs-source
+initramfs-source: $(BUILD_DIR)/initramfs/.config
+	$(MAKE) -C $(BUILD_DIR)/initramfs source
+
+$(BUILD_DIR)/rootfs/.config: $(ROOT_DIR)/buildroot/Makefile
+	$(BR_INIT_ENV) $(MAKE) O=$(BUILD_DIR)/rootfs -C $(ROOT_DIR)/buildroot fischertechnik_TXT_rootfs_defconfig
+
+$(BUILD_DIR)/initramfs/.config: $(ROOT_DIR)/buildroot/Makefile
+	$(BR_INIT_ENV) $(MAKE) O=$(BUILD_DIR)/initramfs -C $(ROOT_DIR)/buildroot fischertechnik_TXT_initramfs_defconfig
+
+$(ROOT_DIR)/buildroot/Makefile:
 	git submodule update --init buildroot
 
-CONFIG_DEPENDS = \
-  .gitmodules \
-  board/fischertechnik/TXT/tisdk_am335x-fischertechnik_txt_defconfig
+$(IMAGE_DIR)/rootfs.img: rootfs
+$(IMAGE_DIR)/uImage: rootfs
+$(IMAGE_DIR)/am335x-kno_txt.dtb: rootfs
+
+.PHONY: rootfs
+rootfs: $(INITRAMFS_DIR)/initramfs.cpio prepare-structure
```

Comment:
I don't see that rootfs uses initramfs.cpio in any way. How does that end up on the sd card?

Reply:
I found it below. I would suggest a comment in the Makefile that initfamfs.cpio is embedded in file uImage (?) via linux_defconfig.

Reply:
> I found it below. I would suggest a comment in the Makefile that initfamfs.cpio is embedded in file uImage (?) via linux_defconfig.

True. I will document that later

Reply:
Maybe the dependency could also be more explicit, for instance that the uImage depends on initramfs.cpio and linux_defconfig.

Reply:
This does not work this way...

Reply:
kernel build in rootfs depends on the existence of the cpio-file...